apps/api/src/controllers/post.ts

```diff
@@ -0,0 +1,130 @@
+import { Request, Response } from 'express';
+import { categories } from './category'
+import { IPost, IPostComment, IPostRequest, IPostResponse } from '../models/interfaces'
+
+export const posts: IPost[] = [];
+const comments: IPostComment[] = [];
+
+export const getPostIndex = (id: string) => {
+  return posts.findIndex((item) => item.id === id);
+};
+
+const getPost = (req: Request, res: Response) => {
+    res.status(200).json(posts);
+};
+
+const createPost = (req: Request, res: Response) => {
+    const { title, image, description, category } = req.body;
+    if (!title) {
+        return res.status(400).json({ message: 'The title is required.' });
+    }
+    if (!image) {
+        return res.status(400).json({ message: 'The image is required.' });
+    }
+    if (!description) {
+        return res.status(400).json({ message: 'The description is required.' });
+    }
+    if (!category) {
+        return res.status(400).json({ message: 'The category is required.' });
+    }
+
+    const newPost: IPost = {
+        id: Date.now().toString(),
+        title,
+        image,
+        description,
+        category,
+        comments: []
+    };
+    posts.push(newPost);
+
+    res.status(201).json(newPost);
+};
+
+const getPostByCategory = (req: Request, res: Response) => {
+    const { categoryId } = req.params;
+
+    const postsByCategory: IPost[] = posts.filter(item => item.category === categoryId);
+
+    postsByCategory.map(postItem => postItem.category = categories.find((categoryItem) => categoryItem.id === postItem.category));
```

Comment:
You can use the method that we exported from category controller `getCategory`, you just need to pass the id of the category.

Also, if you only want to iterate over each post you can use `forEach` instead of `map` since map returns an array but we are not using it.